src/toggler/Toggler.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @overview A module that toggles element visibility
+ * @module Toggler.js
+*/
+
+import feature from 'feature.js';
+
+export const Toggler = {
+  /**
+   * Queries page for Togglers to be initialized, caching them to variables
+   * before calling setup
+   * @param {string} togglerSelector Selector for toggler elements that trigger Togglers
+   * @param {string} toggleableSelector Selector for the element to be toggled
+   * @returns {void}
+  */
+  toggler({togglerSelector, toggleableSelector}) {
+    this.collapsedClass = `is-invisible`;
+    this.togglerEls = document.querySelectorAll(togglerSelector);
+    this.toggleableEls = document.querySelectorAll(toggleableSelector);
+    this.toggleableSelector = toggleableSelector;
+    this._setupTogglers();
+  },
+
+  /**
+   * Sets up all Toggles on the page, adding event listeners to their togglers
+   * @returns {void}
+   * @private
+  */
+  _setupTogglers() {
+    [...this.togglerEls].forEach((el) => {
+      const toggleableEl = el.parentNode.querySelector(this.toggleableSelector);
+      el.setAttribute(`touch-action`, `none`);
+
+      if (feature.touch) {
+        el.addEventListener(`pointerup`, (e) => {
+          e.stopPropagation();
+          this._setupExpandingAndCollapsing(toggleableEl);
+        });
+        this._handleClickOff();
+      } else {
+        el.addEventListener(`pointerenter`, () => {
+          this.expand(toggleableEl);
+        });
+        el.addEventListener(`pointerleave`, () => {
+          this.collapse(toggleableEl);
+        });
+      }
+    });
+  },
+
+  /**
+   * Determines whether to expand (show) or collapse (hide) a given Toggler
+   * @param {object} toggleableEl Toggler DOM element to be expanded or collapsed
+   * @returns {void}
+   * @private
+  */
+  _setupExpandingAndCollapsing(toggleableEl) {
+    if (toggleableEl.classList.contains(this.collapsedClass)) {
+      this.collapseOpen();
+      this.expand(toggleableEl);
+    } else {
+      this.collapse(toggleableEl);
+    }
+  },
+
+  /**
+   * Handles expanding of a given Toggler, collapsing any open Togglers first
```

Comment:
As per the comment, this method is supposed to collapse any open togglers but `expand` method doesn't call `collapseOpen`.